src/app/students/(table)/student-table.tsx

```diff
@@ -38,10 +40,50 @@ const DataTable = <TData, TValue>({
     columns,
     data,
 }: DataTableProps<TData, TValue>) => {
-    const [rowSelection, setRowSelection] = React.useState({})
-    const [columnVisibility, setColumnVisibility] = React.useState<VisibilityState>({})
-    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([])
-    const [sorting, setSorting] = React.useState<SortingState>([])
+    const [rowSelection, setRowSelection] = useState({})
+    const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+    const [sorting, setSorting] = useState<SortingState>([])
+
+    const [pagination, setPagination] = useState<PaginationState>({
+        pageIndex: 1,
+        pageSize: 4,
+    })
+
+    const [pageCount, setPageCount] = useState(0)
+    const totalStudents = useStudents()?.totalStudents ?? 0
+    const router = useRouter()
+    const pathname = usePathname()
+    const searchParams = useSearchParams()
+
+    useEffect(() => {
+        const page = parseInt(searchParams.get('page') ?? '1', 10) - 1
```

Comment:
nit;

`parseInt` defaults to base 10 and most people would assume this, it's kinda just more confusing to specify the `10` because it makes people unnecessarily have to think about this code more than needed

Reply:
This applies to multiple usages of `parseInt()`